### DISCUSSION TITLE
How can i use maigret inside python script?

### DISCUSSION THREAD

**Discussion by krischal** [soxoj/maigret#804] [General]
The only way given is through shell command. But is there any way i can use it through python script?

**Comment by techdragon** [comment]
It is possible but contrary to the documentation saying:

> ## Simple API[](https://maigret.readthedocs.io/en/latest/features.html#simple-api)
>
> Maigret can be easily integrated with the use of Python package [maigret](https://pypi.org/project/maigret/).
> 
> Example: the official [Telegram bot](https://github.com/soxoj/maigret-tg-bot)

It's significantly more complex than it ideally should be and the linked telegram bot is not a particularly good example as its both out of date and bundles its own data to avoid the complexity of using the data built into the maigret library...  

I'm slowly building a more useful python library/wrapper around maigret, but I'm not sure I'll want to share it because of how fragile it will be since a large part of the necessary inputs for the functions is entirely dictated by the argparsing command line functionality.

**Comment by DilLip-Chowdary-Codes** [comment]
@soxoj is there any API we can use instead of cli?

**Comment by soxoj** [comment]
You can import Maigret and utilize the maigret.search function, as demonstrated in [wizard.py](https://github.com/soxoj/maigret/blob/main/wizard.py#L16). This approach is straightforward and provides a list of discovered accounts after completing all the checks.

Please let me know if this meets your needs or if you require a different type of API to retrieve the results. I’ll be happy to implement it.
